crates/bevy_asset/src/loader.rs

```diff
@@ -119,12 +127,26 @@ where
         TypeId::of::<L>()
     }
 
+    fn asset_type_name(&self) -> &'static str {
+        std::any::type_name::<L::Asset>()
+    }
+
     fn asset_type_id(&self) -> TypeId {
         TypeId::of::<L::Asset>()
     }
 
-    fn asset_type_name(&self) -> &'static str {
-        std::any::type_name::<L::Asset>()
+    fn label_type_name(&self, label_type: Option<&str>) -> Option<&'static str> {
```

Comment:
In my opinion this logic should be part of the loader. We could directly call the methods on the loader here.

Reply:
That puts the responsibility of thinking about "if there is no label it's the root Asset type" on the person who implement `AssetLoader`. I think the only reason we might want that is if we want to allow an `AssetLoader` to only take paths with labels, but even if we do that it would still load an asset with a handle with a path without a label so it doesn't really make sense I think?

Reply:
In most cases an asset laoder should not care about the new functions at all. If the default implementations of the functions return the type name/id of the main asset, only the ones requiring their own implementations need to handle "no label" => "main asset type" and is is fine imo. So far I am only aware of the Gltf loader that would need implementations of the new functions.